File: app/src/main/java/com/pkk/andriod/attendence/misc/SharedPref.java
package com.pkk.andriod.attendence.misc;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class SharedPref {

    public void settype(Context context,String userType){
        getEditor(context).putString("type",userType);
    }

    public void gettype(Context context,String userType){
        getEditor(context).putString("type",userType);
    }

    private SharedPreferences.Editor getEditor(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context).edit();
    }

    public void clearPrefs(Activity activity) {
        getEditor(activity).clear().putBoolean("isLoggedIn", false).apply();
    }
}

File: app/src/main/java/com/pkk/android/attendance/misc/SharedPref.kt
package com.pkk.android.attendance.misc

import android.app.Activity
import android.content.Context
import android.content.SharedPreferences
import com.pkk.android.attendance.R

class SharedPref {

    companion object {

        fun setString(context: Context, key: String, value: String) {
            getEditor(context).putString(key, value)
        }

        fun setInt(context: Context, key: String, value: Int) {
            getEditor(context).putInt(key, value)
        }

        fun getString(context: Context, key: String?, def: String = ""): String? {
            return getSharedPreference(context).getString(key, def)
        }

        fun getInt(context: Context, key: String, def: Int = -1): Int {
            return getSharedPreference(context).getInt(key, def)
        }

        private fun getSharedPreference(context: Context): SharedPreferences {
            return context.getSharedPreferences(
                context.getString(R.string.file_preference),
                Context.MODE_PRIVATE
            )
        }

        private fun getEditor(context: Context): SharedPreferences.Editor {
            return getSharedPreference(context).edit()
        }

        fun clearPrefs(activity: Activity) {
            getEditor(activity).clear().apply()
        }

    }
}